### PR TITLE
Backward compatible `key` & `channel` params for config pipeline

### PR DESCRIPTION
**Pull request by qin-yu**
Hi @wolny and @lorenzocerrone 

Since we are about to publish the next plant paper, I made two essential improvements, on the config pipeline, that help user experience a lot and also work with current configs, i.e. `GenericPipelineStep` and its children have been slightly modified for a generic but optional `key`/`channel` parameter for each step in the pipeline, which can be omitted or set to `Null` in the config.

1. optional `channel` selection in each step: for example, in preprocessing, now users can choose a cell/nucleus channel from multichannel images; in segmentation, one can choose a channel from multichannel 3dunet outputs (an additional task during training improves the quality of predictions); same for postprocessing, now can save the right channel to a TIFF file.
2. optional `key` selection in each step: for example, in preprocessing, a dataset can be specified from H5/Zarr files, which have been designed to have multiple datasets; in segmentation step like LiftedMulticut, by specifying a dataset key, one don't have to create file for PlantSeg and can just take a segmentation from H5 files generated by another software.

While the general user experience for such use cases could be greatly. For our paper, I 

- organised data into multi-group/dataset H5 files because each image has multiple raw and label channel, which makes `key` parameter very useful; and 
- trained foreground+boundary networks for bioimage.io and publication, which requires a `channel` parameter otherwise PlantSeg is not even streamlined 

The modified code has been tested with old and new configs (example committed), and I hope this gets reviewed and merged before the publication.

### CODE REVIEW

**Comment by qin-yu** [comment]
This should fix #162

**Comment by qin-yu** [comment]
Here is the update I mentioned @athulrv @tejasvineemody 

Before:

1. Run pytorch-3dunet or plant-seg on raw image for probability maps
2. Select and save boundary prediction channel outside plant-seg
    1. otherwise a wrong channel will be used for segmentation
3. Run plant-seg segmentation on saved channel

Now:

- Run plant-seg on raw image and get segmentation out

**Comment by qin-yu** [comment]
Although this modified code should be backward compatible, to enforce all keyword arguments are passed as keyword arguments (as opposed to positional), @k-dominik proposed a very nice trick of adding a `*, ` before keyword arguments in functions. I shall not make the change here maybe, and let someone change it for the whole software in the future.

File: plantseg/io/tiff.py [hunk 1]
```diff
@@ -166,4 +166,5 @@ def create_tiff(path: str, stack: np.array, voxel_size: list[float, float, float
                      dtype=stack.dtype,
                      imagej=True,
                      resolution=resolution,
-                     metadata={'axes': 'TZCYXS', 'spacing': spacing, 'unit': voxel_size_unit})
+                     metadata={'axes': 'TZCYXS', 'spacing': spacing, 'unit': voxel_size_unit},
+                     compression='zlib')
```
Comment (wolny) [inline plantseg/io/tiff.py:170]: thanks for adding the compression to our tif files!

File: plantseg/pipeline/raw2seg.py [hunk 2]
```diff
@@ -48,20 +53,27 @@ def configure_segmentation_postprocessing_step(input_paths, config):
...
+                                input_type=input_type, output_type=output_type,
                                 save_directory=save_directory, factor=factor, out_ext=out_ext,
                                 state=state, save_raw=save_raw, output_shapes=output_shapes)
 
 
 def _validate_cnn_postprocessing_rescaling(input_paths, config):
-    input_shapes = [load_shape(input_path) for input_path in input_paths]
+    input_key = config["preprocessing"].get('key', None)
+    input_channel = config["preprocessing"].get('channel', None)
+    input_shapes = [load_shape(input_path, key=input_key) for input_path in input_paths]
```
Comment (wolny) [inline plantseg/pipeline/raw2seg.py:74]: nice that finally someone used the `key` argument of this function :)

**Review by wolny — APPROVED** [review]
Really nice work Qin. Looks clean to me. If you tested it and it works then I'm happy to merge. Later on we could open the PR for napari UI to add key/channel to the UI.

**Pull request merged**